### ISSUE TITLE
Make shoulder slope a 'degree' measurement

### ISSUE THREAD

**Issue by joostdecock** [freesewing/freesewing#358]
## Implementation to-do

 - [x] Update models package 
 - [x] Update `neckstimate` method in utils
 - [x] Add `isDegMeasurement` to utils
 - [x] Add support for degree measurement in workbench
 - [x] Port patterns to new `shoulderSlope` measurement
    - [x] aaron
    - [x] benjamin
    - [x] bent
    - [x] breanna
    - [x] brian
    - [x] bruce
    - [x] carlita
    - [x] carlton
    - [x] cathrin
    - [x] diana
    - [x] florence
    - [x] florent
    - [x] holmes
    - [x] huey
    - [x] hugo
    - [x] jaeger
    - [x] penelope
    - [x] sandy
    - [x] shin
    - [x] simon
    - [x] simone
    - [x] sven
    - [x] tamiko
    - [x] theo
    - [x] titan
    - [x] trayvon
    - [x] wahid
    - [x] waralee

## Original issue

With the introduction of Breanna, we changed a bunch of measurements to be better suited to map both the male and female form. We also changed the way we measure shoulder slope, because in practice we noticed that people were routinely measuring this wrong.

I know feel like that was a mistake.

Not that the reasons weren't legit. People really do get their shoulder slope wrong a lot of the time. But the cure might be worse than the disease. We now measure shoulder slope from down at the hip to the shoulder point. This is a very long measurement, and half a cm (or even a cm) more or less is an easy mistake to make when neither the hip line nor the shoulder point itself are things that are obviously marked on a body.

Yet, a centimeter difference in the shoulder slope measurement makes a huge different. This is very obvious when you sample the shoulder slope measurement in our development environment:

![Screenshot_2020-04-11 brian](https://user-images.githubusercontent.com/1708494/79039298-96232d80-7be0-11ea-92bc-9322ae6c4c89.png)

As you can see, the shoulder slope is now the highly volatile measurement that is even harder to get right. So I've made the situation worse.

## Let slopes be slopes

I think in the end, it is just simpler to express a shoulder slope in degrees. After all, that's what it is, a slope. Not a distance.

My first instinct was to not use a shoulder slope measurements anymore, and instead just make it an option with a sensible default that people can adapt.
That has the downside that if you know you have sloped shoulders, you're going to have to set that option manually for each pattern. That's not great.

The solution to that is store the shoulder slope in the model, just like any other measurement, only this one is in degrees. That would work, but has the downside that now everybody has to measure their shoulder slope again, except now it's in degrees, how do you even measure that?

## Proposal

This is a tricky issue, but I think in the end we need to try and make it harder to have mistakes. So I would suggest to convert the shoulder slope measurement to a degree measurement and give everyone the same (sensible) default. A default that they can adapt if they would like that.

Once that's done, we'd need to adapt the patterns to use the degrees, but that will make constructing the shoulder a lot simpler.

**Comment by AlfaLyr** [comment]
Very nice idea. The problem with the way it's calculated now it's that, if you are not very careful and don't maeasure from the same point in the hips for hps to hips and shoulder slope you get inacurate values. A degree is more visual, people can just take an angle transporter, and in the info of the measurement we can even show different drawing with people with different angles so people can see what they look more close to and decide

**Comment by joostdecock** [comment]
> show different drawing with people with different angles

That's a great idea!

**Comment by AlfaLyr** [comment]
I'm glad you like it :) everybody has a protactror around at home to measure angles from a ruller set from school or something, and we can add a drawing explaining how to use it... and whoever can't be bothered to do that can just have a look at the example drawings

**Comment by AlfaLyr** [comment]
I forgot to mention: to avoid the problem of people having to measure themselves again, the new angle could be calculated automatically from the current measurement in an existing model and update it, so no need to change anything if you are happy with your model already

**Comment by AlfaLyr** [comment]
Hi again... I come back because I just thought about a very easy way to calculate the angle: you can just take a selfie standing straight, open the photo editor in your phone, go to the straighten option and rotate the photo until one of your shoulders is horizontal. The angle you rotated the photo is your shoulder slope :)

**Comment by Lobotomik** [comment]
Maybe the issues measuring the shoulder slope come from unclear explanations and illustrations. 

* I find the marble explanation for HPS point a bit obscure. I think it merits some rewording, but even more welcome would be a simple hand drawn illustration, or a 3D mesh, or photo depicting the area (but not in the paper-doll plane).

* The hipline reference is confusing. Other sewing sites and books as well as workout sites consider the hipline to be the horizontal line along the fattest part of your bum. However, freesewing says it is _above the top_ of your hips. Or it says it is where you wear your belt, which depends on the rise your pants. Anyhow, that part of the body is bound to have unwanted flesh introducing much error.

* When measuring HSP to hipline, the text tells you to measure from the HSP straigt to the hipline, but the illustration shows the tape going from the HSP to the _center_ of the hipline (which looks correct). There shouldnn't be much difference, but hey. For shoulder angle, it does say "center of your back at the hipline", but the illustration shows the tape at the tip of the collarbone, a bit far from the shoulder.

I believe the reference points should be easy to find: The hipline interpreted as widest point at the hips is much easier to find than the top of the hipbones, which most people won't know how to find.

Also, the explanations and illustrations could be reviewed to become more clear. I wouldn't mind doing that myself, but I really don't understand them well enough. My own measurements as well as those of other people I've tried with disagree with the prediction in all areas related to hip and waist height, and produce misshapen patterns with weird shoulder slopes, ridiculously short legs and short shirt tails.

A possible solution to the shoulder slope issue would be "What do your shoulders slope like? Click one the picture out of the following 5 that you look most alike".

**Comment by joostdecock** [comment]
This is fully implemented in this repo. There's still changes that are required on the frontend. Please see this issue for those: https://github.com/freesewing/freesewing.org/issues/717

I'll be closing this now, but not without giving some feedback on some of the feedback by @Lobotomik :

**I find the marble explanation for HPS point a bit obscure**

We could probably explain it better, and we will try to improve that. But HPS in general is a fairly standard term among dressmakers, and rather useful, so I think it's not a bad thing to use it as such.

I've been looking into 3D to better illustrate measurements and I've already done some work on that (in particular the plugin that does DXF export for importing into 3D programs). It got a bit stalled because I'm waiting on feedback on the legal aspects of using the 3D avatar that comes with the program. It's not clear whether we can use it, and if we can't we'll need to find an alternative.

Point is: 3D instructions for measurements are on my todo list.

**The hipline reference is confusing**

> Other sewing sites and books as well as workout sites consider the hipline to be the horizontal line along the fattest part of your bum.

Some do. Some don't. Maybe it's a regional thing, or it stems from the fact that most of the online sewing world is dominated by women who rarely measure anything between waist and (what they call) hips.  
For menswear, things are different, and we need that measurement. 

There's no obvious right or wrong here, there's just no consistent way of naming things. So I'm going to not change this.

**HPS to hipline**

> When measuring HSP to hipline, the text tells you to measure from the HSP straigt to the hipline, but the illustration shows the tape going from the HSP to the center of the hipline (which looks correct). There shouldnn't be much difference, but hey.

We're changing that. See https://github.com/freesewing/freesewing/issues/416

**Shoulder slope**

> For shoulder angle, it does say "center of your back at the hipline", but the illustration shows the tape at the tip of the collarbone, a bit far from the shoulder.

This was changed in this issue. Frontend changes are being tracked here: https://github.com/freesewing/freesewing.org/issues/717
